package hector.resources;

import "Scheme.proto";

option optimize_for = SPEED;

message WebSitePath {
	optional string path = 1;			// path, in memory it is stored in the jarray
	optional uint32 path_status = 2;		// status of the item
	optional fixed32 last_path_status_update = 3;	// status last changed
	optional int32 error_count = 4;			// error count
	optional fixed32 cksum = 5;			// checksum
	optional fixed32 last_modified = 6;		// content last changed
	optional fixed32 modification_history = 7;	// 4x1B period of modification (2^n minutes)
}

message WebSiteResource {
	optional Scheme url_scheme = 1;		// HTTP or HTTPS
	required string url_host = 2;		// host name (as seen in URL)
	optional int32 url_port = 3;		// port value
	optional fixed32 ip4_addr = 4;		// IPv4 address (result of DNS query)
	optional fixed64 ip6_addr_1 = 5;	// IPv6 address, part 1
	optional fixed64 ip6_addr_2 = 6;	// IPv6 address, part 2
	optional fixed32 ip_addr_expire = 7;	// IP address DNS mapping expire time

	repeated string allow_urls = 8;	// list of allowed URLs
	repeated string disallow_urls = 9;	// list of disallowed URLs
	optional fixed32 robots_expire = 10;	// when robots.txt will expire
	optional int32 robots_redirect_count = 11;	// number of redirects while fetching robots.txt

	repeated WebSitePath paths = 12;	// all paths
}
